CRM/MembershipExtras/Hook/BuildForm/ContributionEdit.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+class CRM_MembershipExtras_Hook_BuildForm_ContributionEdit {
+
+  /**
+   * @var string
+   *   Path where template with new fields is stored.
+   */
+  private $templatePath;
+
+  public function __construct() {
+    $this->templatePath = CRM_MembershipExtras_ExtensionUtil::path() . '/templates';
+  }
+
+  public function buildForm() {
+    $this->appendModificationTemplate();
+  }
+
+  /**
+   * Appends another template to the
+   * core contribution edit form template
+   * to allow for modifications to
+   * core template.
+   */
+  private function appendModificationTemplate() {
+    CRM_Core_Region::instance('page-body')->add([
```

Comment:
Since this will only have javascript code, do you think it may be better to use something like this instead?

```php
CRM_Core_Resources::singleton()->addScriptFile('uk.co.compucorp.membershipextras', 'js/preventCompletingContribution.js');
```